src/adapter/src/optimize/view.rs

```diff
@@ -52,13 +56,24 @@ impl Optimize<HirRelationExpr> for Optimizer {
         trace_plan!(at: "raw", &expr);
 
         // HIR ⇒ MIR lowering and decorrelation
-        let expr = expr.lower(&self.config, self.metrics.as_ref())?;
+        let mut expr = expr.lower(&self.config, self.metrics.as_ref())?;
 
-        // MIR ⇒ MIR optimization (local)
         let mut df_meta = DataflowMetainfo::default();
         let mut transform_ctx =
             TransformCtx::local(&self.config.features, &self.typecheck_ctx, &mut df_meta);
-        let expr = optimize_mir_local(expr, &mut transform_ctx)?;
+
+        // This optimizer is also used for INSERTs (in addition to VIEWs). Therefore, we first call
+        // a very simple optimizer, which should fully take care of constant INSERTs.
+        expr = optimize_mir_constant_insert(expr, &mut transform_ctx)?;
```

Comment:
I recommend to remove the reference to inserts, and just have `optimize_mir_constants` which does a general thing for all expressions. The comment can call out why it is important (performance on constant inserts).

Reply:
Done